src/uu/ls/src/ls.rs

```diff
@@ -474,6 +510,30 @@ impl Config {
 
         let alloc_size = options.is_present(options::size::ALLOCATION_SIZE);
 
+        let block_size = if std::env::var_os("BLOCK_SIZE").is_some() {
+            if let Some(parsed_block_size) =
+                Config::parse_byte_count(&std::env::var_os("BLOCK_SIZE").unwrap().to_string_lossy())
+            {
+                parsed_block_size
+            } else {
+                DEFAULT_BLOCK_SIZE
+            }
+        } else if std::env::var_os("POSIXLY_CORRECT").is_some() {
+            if std::env::var_os("POSIXLY_CORRECT")
+                .unwrap()
+                .eq(&OsString::from("true"))
+                || std::env::var_os("POSIXLY_CORRECT")
+                    .unwrap()
+                    .eq(&OsString::from("1"))
+            {
+                POSIXLY_CORRECT_BLOCK_SIZE
+            } else {
+                DEFAULT_BLOCK_SIZE
+            }
+        } else {
+            DEFAULT_BLOCK_SIZE
+        };
```

Comment:
```suggestion
       let block_size = if let Ok(size) = std::env::var("BLOCK_SIZE") {
            if let Some(parsed_block_size) = Config::parse_byte_count(&size) {
                parsed_block_size
            } else {
                DEFAULT_BLOCK_SIZE // this branch should probably throw an error
            }
        } else if let Ok(c) = std::env::var("POSIXLY_CORRECT") {
            if c == "true" || c == "1" {
                POSIXLY_CORRECT_BLOCK_SIZE
            } else {
                DEFAULT_BLOCK_SIZE
            }
        } else {
            DEFAULT_BLOCK_SIZE
        };

```

Reply:
I've rearranged this a bit, but the parse_byte_count function is where errors are printed.  Notably, we print a parse error and continue, while GNU does not.

Reply:
Sorry, I don't understand. Are you just talking about the comment in the suggestion or does the suggestion change the behaviour?

Reply:
I'm sorry.  I'm saying a few things things: 1) this is now an Option<usize>, and the code has changed to match the behavior of an Option, and 2) an error is *already printed in this case/branch,* it's just printed in the parse function where the error occurs.